frontend/src/components/CreateMemo/page.tsx

```diff
@@ -100,44 +111,100 @@ const CreateMemo = ({ }) => {
     setReloadDropdown(false);
   };
 
-  const handleSubmit = async (event: any) => {
-    event.preventDefault();
-
-    // TODO: better naming convention for the memo object fields of location
-    const memoToCreate: MemoType = {
-      name: name,
-      date: new Date().toString(),
-      location: {
-        name: locationName,
-        coordinates: [
-          coordinates[0], coordinates[1]
-        ]
-      },
-      description: description,
-      //tags: selectedTags,
+
+  // Function to handle the upload of media files
+  const handleMediaUpload = async (event: React.ChangeEvent<HTMLInputElement>) => {
+      const files = event.target.files;
+      
+      if (files) {
+        const newMedia: Blob[] = Array.from(files);
+        setUploadedMedia(prevMedia => [...prevMedia, ...newMedia]);
+          for (const file of Array.from(files)) {
+              await processMediaFile(file);
+          }
+      }
     };
 
-    try {
-      const res = await MemoController.create_memo(userID, memoToCreate);
+    // Process each media file asynchronously
+    const processMediaFile = async (file: File) => {
+      try {
+          const base64String = await convertBlobToBase64(file);
+          setMedia(prevMedia => [...prevMedia, base64String]);
+      } catch (error) {
+          console.error('Error processing media file:', error);
+      }
+    };
 
-      // console.log(res)
-      const newMemoId = res._id; // TODO
+    // Convert Blob object to Base64-encoded string
+    const convertBlobToBase64 = (blob: Blob): Promise<string> => {
+      return new Promise((resolve, reject) => {
+          const reader = new FileReader();
+          
+          reader.onload = () => {
+              const base64String = reader.result as string;
+              resolve(base64String);
+          };
+          
+          reader.onerror = reject;
+          reader.readAsDataURL(blob);
+      });
+    };
 
+    // Function to create a memo with media
+    const createMemoWithMedia = async (newMedia: Blob[]) => {
+      try {
+          const newMediaStrings: string[] = await Promise.all(newMedia.map(convertBlobToBase64));
+          
+          const memoToCreate: MemoType = {
+              name,
```

Comment:
am i trippin here or is name and description not being set to something?

like
```
{
  name: ... ,
  description: ... ,
  ...
}
```

i can see they are from use state at the top tho

Reply:
yea since in key value pair if I don't define the key name, it will use value variable name as key. So, name and description is just a shorten version of name:name and description:description. I have ran it before and it works

Reply:
oh wow learned something new here